### PR TITLE
fix(link): preserve enabled empty nested object in x-display filter

### PR DESCRIPTION
**Pull request by pxc1130**
## Summary
Refactor OpenAPI response filtering for `x-display`, clarify validation/filter order in HTTP execution flow, and add unit tests for key edge cases.

## Type of Change
- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation update
- [x] Refactoring

## Related Issue
N/A

### 1) Refactor `x-display` response filtering
- Updated: `core/plugin/link/utils/open_api_schema/response_filter.py`
- Replaced path-trie pruning with JSONPath-based filtering via `jsonpath-rw`.
- Kept behavior contract explicit:
  - Remove fields marked with `x-display: false`.
  - If all children under an object/array container are hidden, keep structural type and return empty `{}` / `[]`.

### 2) Adjust HTTP execution flow: validate first, then filter
- Updated: `core/plugin/link/service/community/tools/http/execution_server.py`
- Ensured response schema validation is executed before `x-display` filtering.
- Applied filtering after validation in both request handling and tool debug paths.
- Preserved existing exception handling and telemetry behavior.

### 3) Strengthen schema validation guardrails
- Updated: `core/plugin/link/utils/open_api_schema/schema_validate.py`
- Added defensive handling for non-string `openapi` version values before regex checks.

### 4) Dependency update
- Updated: `core/plugin/link/pyproject.toml`
- Added dependency: `jsonpath-rw>=1.4.0`

### 5) Add focused unit tests
- Added: `core/plugin/link/tests/unit/test_response_filter.py`
- Covered scenarios:
  - Hidden field removal.
  - Hidden field pruning in array items.
  - All-hidden object container returns `{}`.
  - All-hidden array container returns `[]`.

## Testing
- [x] New tests added (unit)
- [ ] Existing full test suite executed
- [ ] Manual testing completed

Test scope added in this PR:
- `test_filter_hidden_field`
- `test_filter_hidden_field_in_array_items`
- `test_all_hidden_object_replaced_with_empty_object`
- `test_all_hidden_array_replaced_with_empty_array`

## Compatibility / Risk
- No API contract change in request shape.
- Response visibility now strictly follows schema `x-display` after validation.
- Low risk: behavior is constrained to response post-processing and covered by unit tests.

## Rollback Plan
If regressions are found:
1. Revert `response_filter.py` implementation to the previous filtering logic.
2. Revert execution order changes in `execution_server.py`.
3. Remove `jsonpath-rw` from dependency list if no longer needed.

## Checklist
- [x] Code follows project coding standards
- [x] Self-review completed
- [x] Documentation/comments updated where needed
- [x] No breaking changes introduced

### CODE REVIEW

File: core/plugin/link/utils/open_api_schema/response_filter.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+    json_path = "$"
+    for token in path_tokens:
+        if token == "[*]":
+            json_path += "[*]"
+        else:
+            json_path += f".{token}"
+    return json_path
+
+
+def _collect_hidden_paths(
```
Comment (zijian2) [inline core/plugin/link/utils/open_api_schema/response_filter.py]: 考虑用jsonpath_rw做正则匹配，示例如下：

```python
from typing import Any, Dict, List, Tuple
from jsonpath_rw import parse

def collect_hidden_paths(schema: Dict[str, Any]) -> Tuple[List[str], Dict[str, str]]:
    """
    使用 jsonpath_rw 收集 schema 中 x-display=false 的路径。

    Returns:
        - need_be_poped_list: list of json paths
        - need_be_emptied_map: dict of json path -> type ('object' or 'array')
    """
    need_be_poped_list: List[str] = []
    need_be_emptied_map: Dict[str, str] = {}

    # JSONPath 查询 x-display=false
    json_expr = '$..[?(@.x-display==False)]'
    matches = parse(json_expr).find(schema)

    for match in matches:
        # match.full_path 返回 jsonpath_rw 类型，需要转换成字符串
        path_str = str(match.full_path)
        need_be_poped_list.append(path_str)

        # 判断是 object 还是 array
        value = match.value
        value_type = value.get("type", "object")
        if value_type in ("object", "array"):
            need_be_emptied_map[path_str] = value_type

    return need_be_poped_list, need_be_emptied_map
```

File: core/plugin/link/utils/open_api_schema/response_filter.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+This module provides utilities for extracting response schema from OpenAPI
+documents and filtering response payloads based on x-display settings.
+"""
+
+from typing import Any, Dict, List, Optional, Tuple
+
+_DELETE = object()
+
+
+class _PathTrieNode:
```
Comment (zijian2) [inline core/plugin/link/utils/open_api_schema/response_filter.py]: 整个节点类都可以用jsonpath_rw实现，无需手动DFS

```python
from typing import Any, Dict, List
from jsonpath_rw import parse

def apply_jsonpath_pruning(
    data: Any,
    delete_paths: List[str],
    empty_paths: Dict[str, str],  # path -> "array" | "object"
) -> Any:
    """Apply delete and replace rules using JSONPath."""
    import copy

    data = copy.deepcopy(data)

    # 删除节点
    for path_expr in delete_paths:
        expr = parse(path_expr)
        matches = expr.find(data)
        for match in matches:
            context = match.context.value
            if isinstance(context, dict):
                context.pop(match.path.fields[0], None)
            elif isinstance(context, list):
                idx = match.path.index
                if 0 <= idx < len(context):
                    context.pop(idx)

    # 置空节点
    for path_expr, kind in empty_paths.items():
        expr = parse(path_expr)
        matches = expr.find(data)
        for match in matches:
            context = match.context.value
            if isinstance(context, dict):
                context[match.path.fields[0]] = {} if kind == "object" else []
            elif isinstance(context, list):
                idx = match.path.index
                if 0 <= idx < len(context):
                    context[idx] = {} if kind == "object" else []

    return data
```

File: core/plugin/link/utils/open_api_schema/response_filter.py (hunk 1 above)
Comment (zijian2) [reply core/plugin/link/utils/open_api_schema/response_filter.py]: path_str = str(match.full_path) 这块不一定要转换，如果后续过滤也是用jsonpath_rw的话

File: core/plugin/link/service/community/tools/http/execution_server.py
Comment (zijian2) [inline core/plugin/link/service/community/tools/http/execution_server.py:1]: 过滤的地方错了，应该只修改 handle_request_execution 和 tool_debug 中结果的处理逻辑，示例如下

```python
result = await http_inst.do_call(span_context)
filter_response_by_x_display(result , openapi_schema)
```
